<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1920</width>
    <height>1080</height>
   </rect>
  </property>
  <property name="maximumSize">
   <size>
    <width>1920</width>
    <height>1080</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_3">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <property name="rightMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QGroupBox" name="menuGroupBox">
        <property name="maximumSize">
         <size>
          <width>300</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="title">
         <string>Menu</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QGroupBox" name="settingsGroupBox">
           <property name="maximumSize">
            <size>
             <width>300</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="title">
            <string>Settings</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_2">
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_4">
                <property name="topMargin">
                 <number>0</number>
                </property>
                <item>
                 <widget class="QLabel" name="puzzleSizeLabel">
                  <property name="text">
                   <string>Puzzle Size</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="horizontalSpacer">
                  <property name="orientation">
                   <enum>Qt::Orientation::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item>
                 <widget class="QSpinBox" name="puzzleSizeSpinBox">
                  <property name="buttonSymbols">
                   <enum>QAbstractSpinBox::ButtonSymbols::UpDownArrows</enum>
                  </property>
                  <property name="minimum">
                   <number>3</number>
                  </property>
                  <property name="maximum">
                   <number>20</number>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_6">
                <property name="topMargin">
                 <number>0</number>
                </property>
                <item>
                 <widget class="QLabel" name="puzzleStateLabel">
                  <property name="text">
                   <string>Puzzle State</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="horizontalSpacer_2">
                  <property name="orientation">
                   <enum>Qt::Orientation::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item>
                 <widget class="QComboBox" name="puzzleStateComboBox">
                  <property name="frame">
                   <bool>true</bool>
                  </property>
                  <item>
                   <property name="text">
                    <string>Solvable</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>Unsolvable</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>Default</string>
                   </property>
                  </item>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_9">
                <property name="topMargin">
                 <number>0</number>
                </property>
                <item>
                 <widget class="QLabel" name="iterationsLabel">
                  <property name="text">
                   <string>Puzzle Iterations</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="horizontalSpacer_3">
                  <property name="orientation">
                   <enum>Qt::Orientation::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item>
                 <widget class="QSpinBox" name="iterationsSpinBox">
                  <property name="buttonSymbols">
                   <enum>QAbstractSpinBox::ButtonSymbols::UpDownArrows</enum>
                  </property>
                  <property name="maximum">
                   <number>100000</number>
                  </property>
                  <property name="value">
                   <number>10000</number>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout">
                <property name="topMargin">
                 <number>0</number>
                </property>
                <item>
                 <widget class="QLabel" name="algorithmLabel">
                  <property name="text">
                   <string>Algorithm</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="horizontalSpacer_4">
                  <property name="orientation">
                   <enum>Qt::Orientation::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item>
                 <widget class="QComboBox" name="algorithmComboBox">
                  <item>
                   <property name="text">
                    <string>Default</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>A*</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>IDA*</string>
                   </property>
                  </item>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_12">
                <property name="topMargin">
                 <number>0</number>
                </property>
                <item>
                 <widget class="QLabel" name="heuristicLabel">
                  <property name="text">
                   <string>Heuristic</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="horizontalSpacer_9">
                  <property name="orientation">
                   <enum>Qt::Orientation::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item>
                 <widget class="QComboBox" name="heuristicComboBox">
                  <item>
                   <property name="text">
                    <string>Manhattan Distance</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>Manhattan Plus LC</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>Hamming Distance</string>
                   </property>
                  </item>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_7">
                <property name="topMargin">
                 <number>0</number>
                </property>
                <item>
                 <widget class="QLabel" name="puzzleGeneratorLabel">
                  <property name="text">
                   <string>Puzzle Generator</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="horizontalSpacer_6">
                  <property name="orientation">
                   <enum>Qt::Orientation::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item>
                 <widget class="QComboBox" name="puzzleGeneratorComboBox">
                  <item>
                   <property name="text">
                    <string>npuzzle-gen.py</string>
                   </property>
                  </item>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </item>
            <item>
             <spacer name="verticalSpacer_2">
              <property name="orientation">
               <enum>Qt::Orientation::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="createPuzzlePushButton">
              <property name="text">
               <string>Create Puzzle</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="choosePuzzlePushButton">
              <property name="text">
               <string>Choose Puzzle File</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="verticalSpacer_4">
              <property name="orientation">
               <enum>Qt::Orientation::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <spacer name="verticalSpacer_3">
              <property name="orientation">
               <enum>Qt::Orientation::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="stopPushButton">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="text">
               <string>Stop</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="startPushButton">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="text">
               <string>Start</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="resultGroupBox">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="title">
            <string>Result</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_6">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_5">
              <property name="topMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QLabel" name="timeToProcessLabel">
                <property name="text">
                 <string>Time to process :</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer_5">
                <property name="orientation">
                 <enum>Qt::Orientation::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QLabel" name="timeToProcessValueLabel">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_10">
              <property name="topMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QLabel" name="numberIterationsLabel">
                <property name="text">
                 <string>Number of iterations :</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer_8">
                <property name="orientation">
                 <enum>Qt::Orientation::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QLabel" name="numberOfIterationsLabel">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_13">
              <property name="topMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QLabel" name="complexityTimeLabel">
                <property name="text">
                 <string>Complexity in time : </string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer_10">
                <property name="orientation">
                 <enum>Qt::Orientation::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QLabel" name="complexityTimeValueLabel">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_8">
              <property name="topMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QLabel" name="complexitySizeLabel">
                <property name="text">
                 <string>Complexity in size : </string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer_7">
                <property name="orientation">
                 <enum>Qt::Orientation::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QLabel" name="complexitySizeValueLabel">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <spacer name="verticalSpacer">
              <property name="orientation">
               <enum>Qt::Orientation::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>292</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="initialStatePushButton">
              <property name="text">
               <string>Initial State</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="goalStatePushButton">
              <property name="text">
               <string>Goal State</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_11">
              <property name="topMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QPushButton" name="backPushButton">
                <property name="text">
                 <string>&lt;-</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="nextPushButton">
                <property name="text">
                 <string>-&gt;</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="puzzleGroupBox">
        <property name="title">
         <string>Puzzle</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <widget class="QWidget" name="puzzleWidget" native="true">
           <layout class="QVBoxLayout" name="verticalLayout_5">
            <item>
             <layout class="QGridLayout" name="puzzleGridLayout"/>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1920</width>
     <height>24</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
